# 2021-PesticideQueens-Bimpatiens

## Contains all analyses performed on datasets for transcriptome analysis in the brain of unmated bumble bee queens exposure by pesticide [(Costa et al. in prep)](https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.15361)

### Synopsis

We used RNA sequencing to explore the sublethal consequences of neonicotinoid pesticide exposure and pollen diet composition in nest-founding bumble bee queens.

### Sample Information

|Queen.ID |	Source.colony |	Pollen.diet |	Pollen.plant.genus | 	Pollen.diet.type |	Pesticide.exposure |	Pesticide.group |
|---------|-------------------|--------------|----------------------|---------------------|-------------------------|-------------------|
|Q085 |	F2 |	diet.1 |	Cistus |	rockrose |	no.exposured |	control |
|Q54B |	F2 |	diet.1 |	Cistus |	rockrose |	no.exposured |	control |
|Q41B |	F3 |	diet.1 |	Cistus |	rockrose |	no.exposured |	control |
|Q57B |	F3 |	diet.1 | Cistus |	rockrose |	no.exposured |	control |
|Q118 |	F4 |	diet.1 |	Cistus |	rockrose |	no.exposured |	control |
|Q71B |	F4 |	diet.1 |	Cistus |	rockrose |	no.exposured |	control |
|Q37B |	F2 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q78B |	F2 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q145B |	F3 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q210 |	F3 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q155B |	F4 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q197 |	F4 |	diet.1 |	Cistus |	rockrose |	early.exposure |	IMD-A |
|Q209 |	F2 |	diet.1 |	Cistus |	rockrose |	chronic.exposure |	IMD-B |
|Q91B |	F2 |	diet.1 |	Cistus |	rockrose |	chronic.exposure |	IMD-B |
|Q225B |	F3 |	diet.1 |	Cistus |	rockrose |	chronic.exposure |	IMD-B |
|Q138B |	F4 |	diet.1 |	Cistus |	rockrose |	chronic.exposure |	IMD-B |
|Q31B |	F2 |	diet.2 |	Erica |	heather |	no.exposured |	control  |
|Q50B |	F2 |	diet.2 |	Erica |	heather	 |no.exposured |	control  |
|Q53B |	F3 |	diet.2 |	Erica |	heather	 |no.exposured |	control  |
|Q63B |	F3 |	diet.2 |	Erica |	heather |	no.exposured |	control  |
|Q29B |	F4 |	diet.2 |	Erica |	heather	 |no.exposured |	control  |
|Q70B |	F4 |	diet.2 |	Erica |	heather |	no.exposured |	control  |
|Q077 |	F2 |	diet.2 |	Erica |	heather	 |early.exposure |	IMD-A |
|Q35B |	F2 |	diet.2 |	Erica |	heather |	early.exposure |	IMD-A |
|Q114 |	F3 |	diet.2 |	Erica |	heather |	early.exposure |IMD-A |
|Q44B |	F3 |	diet.2 |	Erica |	heather	 |early.exposure |	IMD-A |
|Q154B |	F4 |	diet.2 |	Erica |	heather	early.exposure |	IMD-A |
|Q196B |	F4 |	diet.2 |	Erica |	heather	early.exposure |	IMD-A |
|Q122 |	F2 |	diet.2 |	Erica |	heather |	chronic.exposure |	IMD-B |
|Q90B |	F2 |	diet.2 |	Erica |	heather	 |chronic.exposure |	IMD-B |
|Q168B |	F3 |	diet.2 |	Erica |	heather	 |chronic.exposure |	IMD-B |
|Q186 |	F4 |	diet.2 |	Erica |	heather |	chronic.exposure |	IMD-B |
|Q81B |	F4 |	diet.2 |	Erica |	heather |	chronic.exposure |	IMD-B |
|Q55B |	F2 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q76B |	F2 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q180 |	F3 |	combined.diet |	na |	combined |	no.exposured | 	control |
|Q203 |	F3 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q080 |	F4 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q119 |	F4 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q36B |	F4 |	combined.diet |	na |	combined |	no.exposured |	control |
|Q079 |	F2 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q108 |	F2 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q48B |	F2 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q158 |	F3 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q172 |	F3 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q38B |	F3 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q111 |	F4 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q166 |	F4 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q51B |	F4 |	combined.diet |	na |	combined |	early.exposure |	IMD-A |
|Q130 |	F2 |	combined.diet |	na |	combined |	chronic.exposure |	IMD-B |
|Q93B |	F2 |	combined.diet |	na |	combined |	early.exposure |	IMD-B |
|Q101B |	F3 |	combined.diet |	na |	combined |	chronic.exposure |	IMD-B |
|Q141 |	F4 |	combined.diet |	na |	combined |	early.exposure |	IMD-B |
|Q96B |	F4 |	combined.diet |	nav	combined |	chronic.exposure |	IMD-B |


### Alignment details

In brief, the workflow:

````
1. Sequence data quality was evaluated using FASTQC;
2. Sequencing adapters and low quality bases were trimmed/filtered using Trimmomatic;
3. Reads were mapped to the B. impatiens v2.0 reference genome (Sadd et al. 2015) using HiSat2;
	 - Reference genome index was created using OGS annotation (splice sites & exons)
4. Exonic expression (read counts) was extracted and summed for genes using featureCount.
````

### Identification of differentially expressed genes

Differential gene expression was analysed using the DESeq2 package. The 0 day group was excluded from this factorial analysis because these queens were collected on the day of eclosion and did not receive a diet treatment. Expression data from the 0 day group were analysed for a subset of genes. Read counts were transformed using the variance stabilizing transformation with diet, age, colony, or age × diet as factors in the design for sample clustering and visualization of gene expression levels.

_see [DESEq2code.R](https://github.com/claudinpcosta/2020-FatBodyQueens-Bimpatiens/blob/master/DESEq2code.R) for the codes, and we can find the input in the folder [Data](https://github.com/claudinpcosta/2021-PesticideQueens-Bimpatiens/tree/master/DATA)_

### Gene Ontology enrichment analysis

We used GO enrichment analysis to explore gene functions that were overrepresented within the natal colony, age, diet and age × diet DEG lists. In brief:

````````
1.Coding sequences from our B. impatiens gene data set were matched to honey bee (A. mellifera) orthologues in the A. mellifera genome version 4.5 using a blastp reciprocal best match (E-value 1e−6), following the procedure used by Harpur et al. (2017);
2. Previously identified D. melanogaster orthologues were located for these A. mellifera genes in the Ensembl database (Zerbino et al., 2018);
3. Enrichment analysis of “complete” and “slim” GO terms (both biological processes and molecular functions terms) was performed using panther (http://pantherdb.org/tools/compareToRefList.jsp).

`````````

### Analysis of diapause-related gene expression

We also performed a molecular test of the hypothesis that age and diet mediate entry into diapause in bumble bee queens. For this, we performed a Fisher's exact test to identify whether there was a significant degree of overlap between:

_(a) genes differentially expressed between the B. impatiens queens in our study, at age 12 days and fed the 75% sucrose diet (12 days/75%) vs all other samples, and
(b) the B. terrestris queen diapause gene list from [Amsalem et al. (2015)](https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.13410)_
	
	
In brief:

````````
1. Generate gene lists from our B. impatiens data set for this analysis, we performed pairwise Wald tests using DESeq2 (see Identification of differentially expressed genes for codes and inputs);
2. Generate B. terrestris data set, we performed deseq2 with treatment group as the only factor in the model and then used the Wald test to obtain DEGs;
3. Fisher's exact tests of overlap our lists and Amsalem's lists.

`````````

_see [DESeq2BterBimp.R](https://github.com/claudinpcosta/2020-FatBodyQueens-Bimpatiens/blob/master/DESeq2BterBimp.R) for codes and  and we can find the input in the folder [Data](https://github.com/claudinpcosta/2021-PesticideQueens-Bimpatiens/tree/master/DATA)_

